main.cpp

```diff
@@ -75,378 +74,232 @@ send me a DM to check your pull request
-    }
-    else
-    {
-        decline.declinePerson();
-    }
+   
+}
+void GymPlan::removeMember(Person)
+{
+    
 }
+
 /*
  2)
- */
-struct DogWalkerPlan
-{
-    int daysPerWeek = 3;
-    float milesPerWalk = 2.5;
-    bool dogClean = true;
-    unsigned int timesWalked = 22;
 
-    struct Dog
-    {
-        void walkDog()
-        {
+times walked per week
+Notifies walker to start dog walk
 
-        }
+can either jog with the dogs or just walk them
+has a list of dogs to walk
+option to feed dog
 
-        void washDog()
-        {
+ 
+ */
 
-        }
-    };
-    void exerciseDog(int breed, bool dogClean);
+struct Dog
+{
+    bool needsFeeding = false;
+};
 
-    Dog walk, wash;
+
+struct DogWalkerPlan
+{
+    //times walked per week
+    int timesWalkedWeekly = 1;
+    //can either jog with the dogs or just walk them
+    bool involvesJogging = true;
+    
+    std::vector<Dog> dogsToWalk; //has a list of dogs to walk
+    
+    //Notifies walker to start dog walk
+    void notifyWalker();
+    
+    //option to feed dog
+    bool shouldFeedDogs = true;
+    void feedDog(int whichDogToFeed);
+    void feedAllDogs();
 };
-void DogWalkerPlan::exerciseDog(int, bool)
+void DogWalkerPlan::feedDog(int)
 {
-    if(dogClean == true)
-    {
-        walk.walkDog();
-    }
-    else 
-    {
-        wash.washDog();
-    }
+
+}
+void DogWalkerPlan::feedAllDogs()
+{
+
 }
 /*
  3)
+3 things it can do:
+    1) play
+    2) stop
+    3) sync tempo
+
+5 properties:
+    1)is edm
+    2)is rock
+    3)is hiphop
+    4)tempo
+    5)key
  */
 struct DjMix
 {
-    bool inSync = true;
-    int slotHrs = 3;
-
-    unsigned int timesPlayed = 4;
-
-    struct PlayButton
-    {
-        void pressPlay()
-        {
-
-        }
-        void pressTempo()
-        {
-
-        }
-    };
-    void playDj(int slotHrs, bool inSync);
     
-    PlayButton mix, tempo ;
 };
-void DjMix::playDj(int, bool)
-{
-    if(inSync == true)
-    {
-        mix.pressPlay();
-    }
-    else
-    {
-        tempo.pressTempo();
-    }
-}
+
 /*
  4)
+ 3 things it can do:
+    1)start practice
+    2)end practice
+    3)update belt
+
+5 properties:
+    1)is blackbelts
+    2)is whitebelt
+    3)is long session
+    4)is intro
+    5)location number
  */
 struct KaratePractice
 {
-    int timeSlot = 1;
-    int location =4;
-    bool isBoxing = false;
-
-    unsigned int numberOfStudents = 8;
-
-    struct StudentLesson
-    {
-     void takeLesson()
-     {
-
-     }
-     void posponeLesson()
-     {
-
-     }   
-    };
-
-    void practiceKarate(int timeSlot, int location, bool isBoxing);
-
-    StudentLesson karate, boxing;
+    
 };
-void KaratePractice::practiceKarate(int, int, bool)
-{
-    if(isBoxing == false)
-    {
-        karate.takeLesson();
-    }
-    else
-    {
-        boxing.posponeLesson();
-    }
-}
 /*
  5)
+3 things it can do:
+    1)start lesson
+    2)end lesson
+    3)pass student
+
+5 properties:
+    1)is car
+    2)is motorbike
```

Comment:
- [ ] consolidate: Vehicle type.
how about `number of obstacles` or `covers parallel parking`?
- [ ] add another property.